### ISSUE TITLE
Service account token is not refreshed when ApiClient is created with default configuration causing 401 (Unauthorized) error

### ISSUE THREAD

**Issue by crypo** [tomplus/kubernetes_asyncio#253]
I am facing an issue with service account token periodic refresh that was originally implemented in https://github.com/tomplus/kubernetes_asyncio/pull/205/files

**Problem**:

Service account token is getting stale when ApiClient is created with default configuration

**Use case:**

1) Create in-cluster config without parameters:
load_incluster_config()
2) Use default configuration with ApiClient
async with ApiClient() as api:
   v1 = CoreV1Api(api)
   ...
3) Run a job that periodically is making k8s calls 
Expected result: once token is refreshed, all new calls will happen with the new token
Actual result: when token is refreshed, only the current call is using the new token but subsequent calls are using the old token

Analysis:

ApiClient is using a **copy** of default configuration https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/client/api_client.py#L71

When token is refreshed, that same copy is getting updated:
https://github.com/tomplus/kubernetes_asyncio/blob/ad5590c1f7026ddec939c2c0d9a2281cbc05a059/kubernetes_asyncio/client/configuration.py#L359
https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/config/incluster_config.py#L99

Due to the original "default" configuration is not being updated, and that ApiClient is creating a copy of configuration from the default one (that still has the old token), the old original tokens are being sent to kube api server.

Proposed solution:

It is necessary to update the default configuration as well when the token is refreshed (at least in the case when ApiClient uses default configuration) 
A small improvement to the original code can resolve the issue https://github.com/tomplus/kubernetes_asyncio/blob/master/kubernetes_asyncio/config/incluster_config.py#L96

`def load_token_from_file(configuration, *args):
      if self.token_expires_at <= datetime.datetime.now():
        # this will reload the token from disk and update default configuration                
        self.load_and_set()
        configuration.api_key['BearerToken'] = self.token`

Note: this problem might be also actual for https://github.com/kubernetes-client/python

**Comment by crypo** [comment]
Formatting the proposed solution:

```python 
def load_token_from_file(configuration, *args):
    if self.token_expires_at <= datetime.datetime.now(): 
        # this will reload the token from disk and update default configuration
        self.load_and_set() 
        configuration.api_key['BearerToken'] = self.token

**Comment by tomplus** [comment]
Thanks for reporting the issue and your detailed analysis :+1: I'll try to fix it by better initialization of Configuration object.

**Comment by tomplus** [comment]
It's fixed in v24.2.3. Please feel free to reopen if needed.

Official https://github.com/kubernetes-client/python is not affected.
